### PR TITLE
Exercise modal dialog coloring with the help of exercise name.

### PR DESCRIPTION
**Pull request by PerAsperaAdAstra3**
Added the functionality that should make the color-coding of the exercise picker modal dialog possible based on the name of the exercise even if the exerciseId column in the ExerciseInRound table is null. (given that the exercise name corresponds to the name of an actual exercise in the Exercise table.)

### CODE REVIEW

File: src/main/java/training/service/impl/JpaTrainingService.java [hunk 1]
```diff
@@ -112,6 +112,37 @@ public Training edit(Long id, Training training) {
 					}
 					if (!mapExerciseIndexes.containsKey(exerciseInRound.getExerciseId()))
 						mapExerciseIndexes.put(exerciseInRound.getExerciseId(), index.intValue());
+						} else { // Doing the check based on Exercise NAME instead of ID
```
Comment (renatav) [inline src/main/java/training/service/impl/JpaTrainingService.java]: Indent the code! This is really hard to read

File: src/main/java/training/service/impl/JpaTrainingService.java [hunk 2]
```diff
@@ -112,6 +112,37 @@ public Training edit(Long id, Training training) {
...
+						exerciseInRound.getExerciseName();
+						List<Exercise> exerciseListForByNameSearch = exerciseService.findAll();
+						Exercise exerciseOfInterest = new Exercise();
```
Comment (renatav) [inline src/main/java/training/service/impl/JpaTrainingService.java]: Any reason why exerciseOfInterest is instantiated? Why not set it to null initially?

File: src/main/java/training/service/impl/JpaTrainingService.java [hunk 3]
```diff
@@ -112,6 +112,37 @@ public Training edit(Long id, Training training) {
...
+						
+						for(Exercise e : exerciseListForByNameSearch) {
+							if(e.getName().equals(exerciseInRound.getExerciseName())) {
+								exerciseOfInterest = e;
```
Comment (renatav) [inline src/main/java/training/service/impl/JpaTrainingService.java]: Add a break is the only point of the loop is to set exerciseOfIntereset

File: src/main/java/training/service/impl/JpaTrainingService.java [hunk 4]
```diff
@@ -112,6 +112,37 @@ public Training edit(Long id, Training training) {
 					}
 					if (!mapExerciseIndexes.containsKey(exerciseInRound.getExerciseId()))
 						mapExerciseIndexes.put(exerciseInRound.getExerciseId(), index.intValue());
+						} else { // Doing the check based on Exercise NAME instead of ID
+						exerciseInRound.getExerciseName();
+						List<Exercise> exerciseListForByNameSearch = exerciseService.findAll();
+						Exercise exerciseOfInterest = new Exercise();
+						
+						for(Exercise e : exerciseListForByNameSearch) {
```
Comment (renatav) [inline src/main/java/training/service/impl/JpaTrainingService.java]: Why not just write a query to select the exercise? No need for this loop

File: src/main/java/training/service/impl/JpaTrainingService.java (hunk 1 above)
Comment (PerAsperaAdAstra3) [reply src/main/java/training/service/impl/JpaTrainingService.java]: Done

File: src/main/java/training/service/impl/JpaTrainingService.java (hunk 4 above)
Comment (PerAsperaAdAstra3) [reply src/main/java/training/service/impl/JpaTrainingService.java]: Done.

**Pull request merged**